### ISSUE TITLE
[Sourcing][x] [P4]<fdp-approval-flow> Some issues with approval flow

### ISSUE THREAD

**Issue by CSENoni** [SAP/fundamental-ngx#9912]
#### Is this a bug, enhancement, or feature request?

bug

#### Briefly describe your proposal.

**Issue 1:** Add node after becomes afterAll (not consistent with different behaviors)

Case 1: (working fine)

We have two parallel nodes. We add serial node in row 1 first, then add serial node in row 2. It is working fine since there is only 1 on 1 dependency relationship here.

https://github.com/SAP/fundamental-ngx/assets/22842279/488f0bd5-e9c4-49bc-9ead-144d5d866501

Case 2: (fail)

We have one parallel node and one serial node which already has 1 on 1 dependency relationship. We add one parallel node, then add serial node. The added serial node makes a serial node above becomes 1 on 2 dependency. And there is no way to make it looks like case 1 result (1 on 1 only).

https://github.com/SAP/fundamental-ngx/assets/22842279/6e0443b5-9f1c-4517-8ba4-dd6d8b1d9081

We tried to set up configuration for node to disable add node afterAll and beforeAll, but they are not working for those properties.

This can workaround to set empty target in afterNodeAdd.

**Issue 2**: Unable to dnd with any node without status of 'not started'

In our application, for old ui, we could move existing node around with some of our statuses. In GS, with approval-flow component, I believe we cannot move existing node unless we remove and add that node back. The only thing we notice that allow us to move node around is dnd but it is only activated for some conditions, especially with node.status = 'not started'. Could you give us more controls on it?

```
    /** @hidden */
    _isCdkDragDisabled(node: ApprovalGraphNode): boolean {
        return !this._isEditMode || node.blank || node.space || node.status !== 'not started';
    }
```

**Issue 3**: nodeActions are not working properly with custom status:

I believe like issue 2, the node actions are heavily depends on the node.status since we can freely disable/enable any action of node with status of 'not started'. With custom status, I can still enable/disable some actions, but it seems to be not working on disableAddBefore when setting some actions in initial load:

<img width="1253" alt="Screenshot 2023-05-27 at 5 55 09 PM" src="https://github.com/SAP/fundamental-ngx/assets/22842279/b23607ad-7246-4622-8d5f-7be8992b58eb">

This can workaround with setting all nodes' graph meta in onDataReceived. Earlier we could also do that in afterNodeAdd, but there is use case that user exit and it will revert back to initial ApprovalProcess. 

**Issue 4**: node in first row is unclickable after adding second node in second row:

https://github.com/SAP/fundamental-ngx/assets/22842279/88d188ad-b30f-4204-b4f0-9782206fc953

**Issue 5**: there are many 'blank' nodes created along with real nodes, and we have to filter those 'blank' nodes to rebuilds node dependencies before passing to api. It would be nice if you could give only real nodes in valueChanged. 

**Issue 6:** I think this component also needs null check for empty approvalProcess for initial load

<img width="1742" alt="Screenshot 2023-05-27 at 8 50 30 PM" src="https://github.com/SAP/fundamental-ngx/assets/22842279/2268e1c6-00a9-4279-9707-1fca287881c9">

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

Latest and previous.

#### If this is a bug, please provide steps for reproducing it.

bug

#### Please provide relevant source code if applicable.

https://stackblitz.com/edit/angular-y19ktz?file=src%2Fapp%2Fplatform-approval-flow-example.component.ts

**Comment by CSENoni** [comment]
Update some more found issues.

**Comment by droshev** [comment]
@manjunathanagaraj @skaquib123

**Comment by mikerodonnell89** [comment]
Issue 1, case 2 - I think this can be accomplished with click+drag

**Comment by mikerodonnell89** [comment]
Issue 2 it does make sense to me that you can't drag around approvers for a completed status (i.e. approved, rejected, in progress). But I think we need to accommodate for custom statuses here

**Comment by mikerodonnell89** [comment]
Issue 4 I can not recreate - what version of fundamental-ngx are you running?

https://github.com/SAP/fundamental-ngx/assets/2471874/e207530d-3c1d-476b-9de6-afa92ada31df

**Comment by CSENoni** [comment]
@mikerodonnell89 for issue 4, it has to follow exact step like this:

https://github.com/SAP/fundamental-ngx/assets/22842279/83d63a6d-decb-4ebd-b0ed-3be0adcc6c48

For issue 1, I am not sure how to do that, but even that works, it may not good for us to tell user to do it in a different way for that specific case.

**Comment by droshev** [comment]
> , it has to follow exact step like this:

@CSENoni Issues related to the approval flow can be addressed to @manjunathanagaraj and his team.

**Comment by CSENoni** [comment]
Hello, could you at least fix the issue 2 for us to give us more controls on which nodes can be dragged/dropped/move? As described, in our application, we allow user to drag/drop/move node for some statuses even it is in progress. Now, we have around 10 customers request us to have this one working.

**Comment by CSENoni** [comment]
Quick question: what does this "Add to next serial node" do when we try to add parallel node? It looks to me it does not do anything. Sometimes it just add a new parallel node and sometimes it does not add anything.

https://github.com/SAP/fundamental-ngx/assets/22842279/d363f0ce-a59e-45fe-bb75-62bfbe49685b

**Comment by skaquib123** [comment]
@CSENoni I have checked this "Add to next serial node" checkbox and it doesn't seem to have any effective functionality. I will remove it or make it optional.

**Comment by droshev** [comment]
According to the team the issue has been resolved => closing it

**Comment by CSENoni** [comment]
I don't see anything fix from `0.43.9`. All issues are still there. Now, we have to override some exposed functions and style to workaround for that so we may not need it. But for something like localization which was addressed in another issue and "Add to next serial node" -> please fix that as stated. Thanks!
